beacon-chain/db/filesystem/blob.go

```diff
@@ -45,11 +37,11 @@ type BlobStorageOption func(*BlobStorage) error
 // WithBlobRetentionEpochs is an option that changes the number of epochs blobs will be persisted.
 func WithBlobRetentionEpochs(e primitives.Epoch) BlobStorageOption {
```

Comment:
If `WithBlobRetentionEpochs` is not invoked, the pruner will not initialize, correct? It appears that `WithBlobRetentionEpochs` is intended solely for altering the number of epoch blobs to retain, but it also seems to initialize the pruner internally. Is this ok?

Reply:
This always runs, with a default value if the flag was not specified: https://github.com/prysmaticlabs/prysm/blob/b45a6664beeb169f9b1b6509930eaab9f3edb208/cmd/beacon-chain/storage/options.go#L24

I did consider making this a required argument to NewBlobStorage, but my thinking was that we may want the pruner to not run at all for full archive nodes, that's why I added the log.Warn if it is not initialized. If we want to make it a required argument, we could add a special flag for nodes that should *never* prune and use that to completely disable the pruner.